core/services/fluxmonitor/flux_monitor_test.go

```diff
@@ -631,6 +640,187 @@ func TestPollingDeviationChecker_RoundTimeoutCausesPoll_timesOutAtZero(t *testin
+
+	tests := []struct {
+		name             string
+		timeout          uint64
+		expectedToSubmit bool
+	}{
+		{"round will time out", 2, true},
+		{"round will not time out", 100, false},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			fluxAggregator := new(mocks.FluxAggregator)
+
+			fluxAggregator.On("SubscribeToLogs", mock.Anything).Return(true, eth.UnsubscribeFunc(func() {}), nil)
+			fluxAggregator.On("GetOracles").Return(oracles, nil)
+
+			// 1st call to roundState in resetTickers
+			fluxAggregator.On("RoundState", nodeAddr, uint32(0)).Return(contracts.FluxAggregatorRoundState{
+				ReportableRoundID: 1,
+				EligibleToSubmit:  false,
+				LatestAnswer:      answerBigInt,
+				StartedAt:         now(),
+				PaymentAmount:     paymentAmount,
+				AvailableFunds:    availableFunds,
+				Timeout:           test.timeout,
+			}, nil).Once()
+
+			// 2nd roundstate call means round timer triggered
+			chRoundState := make(chan struct{})
+			fluxAggregator.On("RoundState", nodeAddr, uint32(0)).Return(contracts.FluxAggregatorRoundState{
+				ReportableRoundID: 1,
+				EligibleToSubmit:  false,
+				LatestAnswer:      answerBigInt,
+				StartedAt:         now(),
+				PaymentAmount:     paymentAmount,
+				AvailableFunds:    availableFunds,
+				Timeout:           0,
+			}, nil).
+				Run(func(mock.Arguments) { close(chRoundState) }).
+				Maybe()
+
+			deviationChecker, err := fluxmonitor.NewPollingDeviationChecker(
+				store,
+				fluxAggregator,
+				logBroadcaster,
+				initr,
+				nil,
+				runManager,
+				fetcher,
+				nil,
+				func() {},
+			)
+			require.NoError(t, err)
+
+			deviationChecker.Start()
+			deviationChecker.OnConnect()
+
+			if test.expectedToSubmit {
+				gomega.NewGomegaWithT(t).Eventually(chRoundState).Should(gomega.BeClosed())
+			} else {
+				gomega.NewGomegaWithT(t).Consistently(chRoundState).ShouldNot(gomega.BeClosed())
+			}
+
+			deviationChecker.Stop()
+			fluxAggregator.AssertExpectations(t)
+		})
+	}
+}
+
+func TestPollingDeviationChecker_UsesPreviousRoundStateOnStartup_IdleTimer(t *testing.T) {
+	store, cleanup := cltest.NewStore(t)
+	defer cleanup()
+
+	nodeAddr := ensureAccount(t, store)
+	fetcher := new(mocks.Fetcher)
+	runManager := new(mocks.RunManager)
+	logBroadcaster := new(mocks.LogBroadcaster)
+
+	job := cltest.NewJobWithFluxMonitorInitiator()
+	initr := job.Initiators[0]
+	initr.PollTimer.Disabled = true
+	initr.IdleTimer.Disabled = false
+
+	answerBigInt := big.NewInt(100)
+	paymentAmount := store.Config.MinimumContractPayment().ToInt()
+	availableFunds := big.NewInt(1).Mul(paymentAmount, big.NewInt(1000))
+
+	almostExpired := time.Now().
+		Add(initr.IdleTimer.Duration.Duration() * -1).
+		Add(2 * time.Second).
+		Unix()
+
+	tests := []struct {
+		name             string
+		startedAt        uint64
+		expectedToSubmit bool
+	}{
+		{"heartbeat about to expired", uint64(almostExpired), true},
+		{"heartbeat will not expire", 100, false},
```

Comment:
1) Can we use "idle timer" or "poll timer"? I think heartbeat is a bit vague.
2) Can we add a test case for `startedAt` = 0? That is a specific condition I'm concerned with.